### PR TITLE
feat(MenuToggle/Select): add status

### LINKED ISSUE #10025: Bug - Select - validated
The v4 `Select` supported a `validated` property; however, it has been omitted for v5. The [design guidelines](https://www.patternfly.org/components/menus/select/design-guidelines#select-lists-with-validation) currently state the "Select component offers three validation-state presentations", but I don't see how the ["validated selections" example](https://www.patternfly.org/components/menus/select/react-deprecated#validated-selections) is achieved in v5?

We've been using validation in Cost Management, but are blocked from updating to the v5 `Select`.

**Cost Management snapshot**
Screenshot 2024-01-19 at 9 19 55 AM

### PR DESCRIPTION
**Pull request by kmcfaul**
**What**: Closes #10025

* Adds `status` and `statusIcon` to `MenuToggle`
* Adds MenuToggle and Select example for new prop

### CODE REVIEW

**Review by thatblindgeye — APPROVED** [review]
This looks good. Only outstanding thing is adding something along the lines of the following to the example description:

>When the status value is "warning" or "danger", you must include helper text that conveys what is causing the warning/error.

This is something that would be covered by a11y docs as well, though, so may not be totally necessary to include it here. Let me know what you think, but not really blocking to me.

**Comment by kmcfaul** [comment]
I think it's a good add for consistency. Also added HelperTexts to the menutoggle example too.

**Review by mcoker — APPROVED** [review]
🚀

File: packages/react-core/src/components/MenuToggle/examples/MenuToggle.md [hunk 1]
```diff
@@ -262,4 +266,15 @@ To create a typeahead toggle, pass in `variant="typeahead"` to the `<MenuToggle>
 To create a multiple typeahead toggle, pass a `<TextInputGroup>` component implemented like the [text input group's filter example](/components/text-input-group#filters) as a child of `<MenuToggle>`.
 
 ```ts file='MenuToggleTypeahead.tsx'
+
+```
+
+### Status toggle
+
+To create a toggle with a status, pass in `status` to the `MenuToggle`. The default icon associated with each status may be overridden by using the `statusIcon` property.
```
Comment (tlabaj) [inline packages/react-core/src/components/MenuToggle/examples/MenuToggle.md:274]:
```suggestion
To create a toggle with a status, pass in the `status` property to the `MenuToggle`. The default icon associated with each status may be overridden by using the `statusIcon` property.
```

**Review by tlabaj — APPROVED** [review]

File: packages/react-core/src/components/MenuToggle/MenuToggle.tsx [hunk 2]
```diff
@@ -37,6 +46,10 @@ export interface MenuToggleProps
   splitButtonOptions?: SplitButtonOptions;
   /** Variant styles of the menu toggle */
   variant?: 'default' | 'plain' | 'primary' | 'plainText' | 'secondary' | 'typeahead';
+  /** Status styles of the menu toggle */
+  status?: 'success' | 'warning' | 'danger';
```
Comment (tlabaj) [inline packages/react-core/src/components/MenuToggle/MenuToggle.tsx:50]: Examples are marked beta.  Can we mark these props beta please.

**Review by wise-king-sullyman — APPROVED** [review]

**Pull request merged**